source/screens/LoginScreen.js

```diff
@@ -138,9 +144,26 @@ function LoginScreen(props) {
   } = useContext(AuthContext);
   const showNotification = useNotification();
 
-  const [modalVisible, setModalVisible] = useState(false);
+  const [loginModal, setLoginModalVisible] = useState(false);
+  const [userAgreementModal, setUserAgreementModalVisible] = useState(false);
```

Comment:
Feels more consistent with loginModalVisible and userAgreementModalVisible maybe? But the names are getting a bit too long.